imap_processing/codice/codice_l1a.py

```diff
@@ -719,6 +789,75 @@ def get_params(dataset: xr.Dataset) -> tuple[int, int, int, int]:
     return table_id, plan_id, plan_step, view_id
 
 
+def group_data(packets: xr.Dataset) -> list[bytes]:
+    """
+    Organize continuation packets into appropriate groups.
+
+    Some packets are continuation packets, as in, they are packets that are
+    part of a group of packets. These packets are marked by the `seq_flgs` field
+    in the CCSDS header of the packet. For CoDICE, the values are defined as
+    follows:
+
+    3 = Packet is not part of a group
+    1 = Packet is the first packet of the group
+    0 = Packet is in the middle of the group
+    2 = Packet is the last packet of the group
+
+    For packets that are part of a group, the byte count associated with the
+    first packet of the group signifies the byte count for the entire group.
+
+    Parameters
+    ----------
+    packets : xarray.Dataset
+        Dataset containing the packets to group.
+
+    Returns
+    -------
+    grouped_data : list[bytes]
+        The packet data, converted to bytes and grouped appropriately.
+    """
+    grouped_data = []  # Holds the properly grouped data to be decompressed
+    current_group = bytearray()  # Temporary storage for current group
+    group_byte_count = None  # Temporary storage for current group byte count
+
+    for packet_data, group_code, byte_count in zip(
+        packets.event_data.data, packets.seq_flgs.data, packets.byte_count.data
+    ):
+        # Convert from numpy array to byte object
+        compressed_packet_values = ast.literal_eval(str(packet_data))
+
+        # If the group code is 3, this means the data is not part of a group
+        # and can be decompressed as-is
+        if group_code == 3:
+            values_to_decompress = compressed_packet_values[:byte_count]
+            grouped_data.append(values_to_decompress)
+
+        # If the group code is 1, this means the data is the first data in a
+        # group. Also, set the byte count for the group
+        elif group_code == 1:
+            group_byte_count = byte_count
+            current_group = compressed_packet_values
+
+        # If the group code is 0, this means the data is part of the middle of
+        # the group
+        elif group_code == 0:
+            current_group += compressed_packet_values
+
+        # If the group code is 2, this means the data is the last data in the
+        # group
+        elif group_code == 2:
+            current_group += compressed_packet_values
+            values_to_decompress = current_group[:group_byte_count]
```

Comment:
Do you know if there are any that are longer than the group_byte_count that are getting trimmed by this? Do we need to warn/log that we got more data than expected in the packets?